frontend/class-frontend.php

```diff
@@ -2011,42 +2015,106 @@ private function fetch_social_profiles() {
 	 *
 	 * @since 1.5.7
 	 *
-	 * @link  https://developers.google.com/webmasters/structured-data/slsb-overview
+	 * @link  https://developers.google.com/structured-data/site-name
 	 */
-	public function internal_search() {
+	public function website() {
+		$output = '{ "@context": "http://schema.org",
+			"@type": "WebSite",
+			"url": "' . esc_attr( $this->get_home_url() ) . '",
+			"name": "' . esc_attr( $this->get_website_name() ) . '"';
+
+		$output .= $this->add_alternate_name();
+		$output .= $this->internal_search_section();
+
+		$output .= '}';
+
+		$this->json_ld_output( $output );
+	}
+
+	/**
+	 * Retrieves the home URL
+	 *
+	 * @return string
+	 */
+	private function get_home_url() {
+		return trailingslashit( home_url() );
+	}
+
+	/**
+	 * Returns an alternate name if one was specified in the WP SEO settings
+	 *
+	 * @return string
+	 */
+	private function add_alternate_name() {
+		if ( '' !== $this->options['alternate_website_name'] ) {
+			return ',"alternateName": "' . esc_attr( $this->options['alternate_website_name'] ) . '"';
+		}
+
+		return '';
+	}
+
+	/**
+	 * Adds the internal search JSON LD code if it's not disabled
+	 *
+	 * @link https://developers.google.com/structured-data/slsb-overview
+	 *
+	 * @return string
+	 */
+	private function internal_search_section() {
+		if ( ! is_front_page() ) {
+			return '';
+		}
+
 		/**
 		 * Filter: 'disable_wpseo_json_ld_search' - Allow disabling of the json+ld output
 		 *
 		 * @api bool $display_search Whether or not to display json+ld search on the frontend
 		 */
-		if ( apply_filters( 'disable_wpseo_json_ld_search', false ) === true ) {
-			return;
-		}
-		$home_url = trailingslashit( home_url() );
+		$disable_json_ld_search = apply_filters( 'disable_wpseo_json_ld_search', false );
```

Comment:
Yeah I purposely didn't do that as it made it harder to read because we're checking for false, not for true.